### PR TITLE
Improved error handling on submit

### PR DESCRIPTION
**Pull request by srh-sloan**

### CODE REVIEW

File: api/routes/application/routes.py [hunk 1]
```diff
@@ -121,9 +123,23 @@ def submit(self, application_id):
...
                 "email": account.email,
             }, 201
         except KeyError as e:
             return {"code": 404, "message": str(e)}
+        except NotificationError as e:
+            current_app.logger.exception(
+                "Notification error on sending SUBMIT notification for"
+                f" application {application_id}"
+            )
+            # return {"code": 500, "message": str(e)}
```
Comment (Adam-W1) [inline api/routes/application/routes.py:137]: think this is a hangover 👀  (also line 144)

File: api/routes/application/routes.py [hunk 2]
```diff
@@ -121,9 +123,23 @@ def submit(self, application_id):
...
+            return str(e), 500, {"x-error": "notification error"}
+        except Exception as e:
+            current_app.logger.exception(
+                "Error on sending SUBMIT notification for application"
+                f" {application_id}"
+            )
+            # return {"code": 500, "message": str(e)}
```
Comment (gio-karunakaran) [inline api/routes/application/routes.py:144]: should we remove this aswell?

**Review by gio-karunakaran — APPROVED** [review]
LGTM :) - minor comment

File: api/routes/application/routes.py (hunk 1 above)
Comment (srh-sloan) [reply api/routes/application/routes.py]: Thanks - both gone

**Pull request merged**